src/frontend/screens/CategoryChooser.js

```diff
@@ -71,20 +71,52 @@ const Item = React.memo(
 const CategoryChooser = ({ navigation }: { navigation: NavigationProp }) => {
   const [{ presets }] = useContext(ConfigContext);
   const [{ value: draftValue }, { updateDraft }] = useDraftObservation();
-
   const presetsList = Array.from(presets.values())
     // Sort presets by sort property and then by name, then filter only point presets
     .sort(presetCompare)
     // Only show presets where the geometry property includes "point"
     .filter(p => p.geometry.includes("point"));
-
   const handleSelectPreset = (preset: Preset) => {
-    updateDraft({
-      tags: {
-        ...(draftValue || {}).tags,
-        categoryId: preset.id,
-      },
-    });
+    const newTags = preset.tags;
+    if (Object.keys(draftValue.tags).length > 0) {
+      // We're updating an observation
+      // Get tags from old category
+      const oldTags = draftValue.tags;
+      const oldCategoryTags = presetsList.filter(
+        p => p.id === oldTags.categoryId
+      )[0].tags;
```

Comment:
```suggestion
      const oldCategoryTags = (presets[oldTags.categoryId] || {}).tags || {}
```
Much simpler :), but it's important to handle the case where there is not a preset matching the categoryId, or where categoryId is undefined (this suggestion will do that - are you using Flow to check this? I helps catch stuff like this)